Online Documentation (Read The Docs)

daleeidd opened the pull request:
Status: opening for feedback. But it is close to merging.

Adds online documentation which will be hosted by [Read the Docs](https://readthedocs.org).

Special thanks to @Nikhil-Work for starting this process.

It can already be view at: https://crest.readthedocs.io/en/latest

Adding a parameter will preload the tab (eg rp=urp).
https://crest.readthedocs.io/en/latest/user/getting-started.html?rp=birp
https://crest.readthedocs.io/en/latest/user/getting-started.html?rp=hdrp
https://crest.readthedocs.io/en/latest/user/getting-started.html?rp=urp

It has merged documentation from all pipelines into one set of online documentation. Pipeline specific content is displayed using tabs, labels or extra sections. PDFs can be generated where pipeline specific content is be stripped.

Here are the PDFs that will be generated when running `make pdf`:
[crest-birp.pdf](https://github.com/wave-harmonic/crest/files/6176829/crest-birp.pdf)
[crest-hdrp.pdf](https://github.com/wave-harmonic/crest/files/6176830/crest-hdrp.pdf)
[crest-urp.pdf](https://github.com/wave-harmonic/crest/files/6176832/crest-urp.pdf)

Here is the PDF generated by RTDs (all pipelines in one. Will likely disable this):
https://crest.readthedocs.io/_/downloads/en/latest/pdf/

I have removed content from USERGUIDE.md leaving only what I have yet to transfer.

There is one sentence per line which is nice for source control.

I also made some improvements here or there. I tried leveraging Unity's documentation more.

Let me know what you think.

## Usage

See the docs/README.md file

daleeidd commented:
Thanks! I have done a pass for dark theme compatibility. Hopefully I go everything.

Nikhil-Work commented:
There is some text in the search window that is blacked out. (at the bottom)
![image](https://user-images.githubusercontent.com/53100819/110691798-98588800-81b3-11eb-9155-3a59effc6a8e.png)

When the search window highlights text it gets whited out.
![image](https://user-images.githubusercontent.com/53100819/110691820-a0182c80-81b3-11eb-87b5-7518aeffc3ea.png)

That's all I could find. It looks great!

huwb commented:
While I'm not intimately familiar with all the bits,. It looks great to me from Chrome on my phone. Nicely done!!

daleeidd commented on docs/user/faq.rst:
```diff
@@ -99,3 +99,27 @@ How to implement a swimming character?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+This is tricky because the underwater effect uses the opaque scene depths in order to render the water fog, which will not include transparents.
+
+.. only:: birp or urp
```
I just pushed a README.md which documents this all better. The birp, hdrp and urp tags only apply to the PDFs we generate. So anything using these will always show for HTML.

In this case I went with using tabs.

huwb replied:
Ok thanks!

daleeidd commented:
> While I'm not intimately familiar with all the bits,. It looks great to me from Chrome on my phone. Nicely done!!

Thanks. I've added a parameter so we can preselect a tab. Details in original post.

I do have a question about the version history. SRP has version going back to 2, but URP only goes back to 3. And it looks like URP version 3.x has changes from the SRP 2.x in them. I've made note of this in the history.rst file. Is the SRP readme.tex incorrect?

daleeidd commented:
Done another pass.

I found this about the water body wizard:
> It is available on the *master* branch in this repository, but not yet in Crest URP/HDRP.

But it is there in URP from the asset store. I couldn't see it in the release notes. Should we add it 4.9 release notes as preview?

huwb commented:
> Should we add it 4.9 release notes as preview

Yes, thanks!

huwb commented:
Thanks for this. Just a few thoughts:

* Should the individual versions be recorded from the table of contents? It makes the list long
* I'd probably expect faq questions to not be individually listed in toc too, but lmk if this was intentional
* Also wondering if toc should be three levels deep (2.1.1) or just two levels deep (2.1). I guess material params stand out as not being necessary for individual listing on the contents. However it's fine if they are.
* Switch to Linear space rendering - this does not apply to hdrp which is linear only

Those are the only things that I could find. Generally looks awesome. Amazing work!

While deeply reviewing small tweaks came to mind so I added them to the branch, hope that's ok and feel free to revert or change them if needed.

huwb commented:
Btw it occurred to me i might be over-weighting the TOC length issue, but it's the first thing the user sees when they click the documentation link, and i first did it on my phone so was presented with a long list, without the benefit of the distilled lists on the left/right panes that you see from desktop.

This does make me wonder if users landing on a raw TOC list is ideal, and wondered if TOC should be added to the introduction section which gives context, social links, videos and is a nicer place to land. It also feels like the TOC are off the grid a bit - when i'm in some section of the documentation the TOC is not listed in teh navigation pane on the left, and you have to click the crest logo to get back there.

I could 100% understand if this is "just how RTD works", but just throwing it out there!

huwb commented:
None of the above are merge blockers in my eyes so let me know how you want to proceed

daleeidd commented:
> * Should the individual versions be recorded from the table of contents? It makes the list long
> * I'd probably expect faq questions to not be individually listed in toc too, but lmk if this was intentional
> * Also wondering if toc should be three levels deep (2.1.1) or just two levels deep (2.1). I guess material params stand out as not being necessary for individual listing on the contents. However it's fine if they are.

Agreed. I was having issues since Sphinx is limited and buggy. I had to hack in some Latex commands and CSS to get this to work. Made a breakthrough at the 11th hour.

> * Switch to Linear space rendering - this does not apply to hdrp which is linear only

Fixed. Thanks.

> While deeply reviewing small tweaks came to mind so I added them to the branch, hope that's ok and feel free to revert or change them if needed.

Nice.

> Btw it occurred to me i might be over-weighting the TOC length issue, but it's the first thing the user sees when they click the documentation link, and i first did it on my phone so was presented with a long list, without the benefit of the distilled lists on the left/right panes that you see from desktop.
> 
> This does make me wonder if users landing on a raw TOC list is ideal, and wondered if TOC should be added to the introduction section which gives context, social links, videos and is a nicer place to land. It also feels like the TOC are off the grid a bit - when i'm in some section of the documentation the TOC is not listed in teh navigation pane on the left, and you have to click the crest logo to get back there.
> 
> I could 100% understand if this is "just how RTD works", but just throwing it out there!

Agreed. I did try this in the beginning but it gets a little buggy and weird. So it will take a bit of time to work out. I'll have another look after 4.9.

---

There is a bug in the PDF where **ABOUT** is the heading of the ToC, but I will have to try and fix that another time.

Also, be wary that the styles for the site might not always update as they are cached. I'll have to add invalidate the cache but will be okay for now.

daleeidd commented:
I've updated the PDFs in OP. Will merge in and merge across.

huwb commented:
Great stuff!
